### PR TITLE
[IMP] event: ease registration follow-up 

### PR DESCRIPTION
**Pull request by pyduf**
Globally ease event registrations followup by improving the related search
views and by allowing a quick access to statistics.

Specifications:

1. A stat button has been added on event that go to attendee reporting filtered
on the event

2. a) in the attendee reporting, the text search has been modified as follows:
- order: first search on event
- order + added: then search on responsible
- order + added: then search on organizer
- order: then search on participant
- then other field already present

2. b) still in the attendee reporting, filter "archived" has been moved as the
last filter

2. c) still in the attendee reporting, group by for campaign, medium and source
have been added

3. link to confirmed and expected attendee in the kanban view box now redirect
to the new stat view with the suitable filter

Technical remarks:
- view for a specific event launched from the event view uses a specific
search view which doesn't include search on event (name, organizer, ...).

Task 2761011

### CODE REVIEW

File: addons/event/views/event_event_views.xml [hunk 1]
```diff
@@ -0,0 +1,345 @@
+<?xml version="1.0"?>
+<odoo>
```
Comment (tde-banana-odoo) [inline addons/event/views/event_event_views.xml:2]: As anyway we break the history, we can try to gain an indent level on all view definitions by "shadowing" the data one, like
```
<odoo><data>
    <record>
        ...
    </record>
</data></odoo>
```
That's not a full xml level based indent but that's commonly done to avoid loosing too much indent levels :)

File: addons/event/__manifest__.py [hunk 2]
```diff
@@ -23,7 +23,10 @@
         'security/ir.model.access.csv',
         'views/event_menu_views.xml',
         'views/event_ticket_views.xml',
-        'views/event_views.xml',
+        'views/event_mail_views.xml',
```
Comment (tde-banana-odoo) [inline addons/event/__manifest__.py:26]: First commit should be a [MOV] (ref is more when we refactor and change code / feature). You can also explain why we do it: to have one file per model, easing finding of views and following future changes, shorter files, ...

File: addons/event/views/event_registration_views.xml [hunk 3]
```diff
@@ -335,4 +379,4 @@
             action="action_registration"
             groups="event.group_event_user"/>
     </data>
-</odoo>
\ No newline at end of file
+</odoo>
```
Comment (tde-banana-odoo) [inline addons/event/views/event_registration_views.xml:382]: Note that an end-of-file change is done in commit 2 while it should be done in commit 1, just for "perfect commits split" purpose :)

File: addons/event/views/event_event_views.xml [hunk 4]
```diff
@@ -342,4 +350,4 @@
             parent="event.menu_reporting_events"
             groups="event.group_event_user"/>
     </data>
-</odoo>
\ No newline at end of file
+</odoo>
```
Comment (tde-banana-odoo) [inline addons/event/views/event_event_views.xml:353]: Same here about "'end of file" change :)

File: addons/event/views/event_registration_views.xml [hunk 5]
```diff
@@ -247,11 +247,32 @@
...
+                                context="{'group_by': 'utm_medium_id'}"/>
+                        <filter string="Source" name="group_by_utm_source_id" domain="[]"
+                                context="{'group_by': 'utm_source_id'}"/>
                    </group>
                 </search>
             </field>
         </record>
 
+        <!-- Search across multiple events, add event search (name, organizer, ...) -->
+        <record id="view_registration_search_not_specific_to_an_event" model="ir.ui.view">
```
Comment (tde-banana-odoo) [inline addons/event/views/event_registration_views.xml:262]: If you define another view, you have to give priorities to clearly state which one is the default one. Odoo takes the one with the lowest priority as default.

File: addons/event/views/event_registration_views.xml [hunk 6]
```diff
@@ -209,7 +209,7 @@
         </record>
 
         <record model="ir.ui.view" id="view_registration_search">
-            <field name="name">event.registration.search</field>
+            <field name="name">event.registration.search.specific.to.an.event</field>
```
Comment (tde-banana-odoo) [inline addons/event/views/event_registration_views.xml:212]: I think I would do the inverse move: keep the default view as it is (that way we don't break anything in custom modules) and define an ``event_registration_view_search_event_specific`` that inherit from the generic one and remove the event_id fields (and set it as less important, for example priority 32). Then I would use that specific one in the ``act_event_registration_stat_from_event`` .

That way for anything else that the new action, nothing changes and we keep behavior as previous unless specifically coming from an event.

File: addons/event/views/event_registration_views.xml [hunk 7]
```diff
@@ -318,6 +339,29 @@
...
+            <field name="help" type="html">
+                <p class="o_view_nocontent_smiling_face">
+                    No Attendees yet!
+                </p><p>
+                    From this dashboard you can report, analyze and detect trends regarding your event registrations.
+                </p>
+            </field>
+        </record>
+
+	    <record id="act_event_registration_stat_from_event" model="ir.actions.act_window">
```
Comment (tde-banana-odoo) [inline addons/event/views/event_registration_views.xml:352]: According to guidelines this should be ``event_registration_action_stats_from_event`` (model, purpose, details).

**Review by tde-banana-odoo — COMMENTED** [review]
Validation ! Thanks for the split, better to spot differences now :) added some new comments, mainly guidelines / organization

**Comment by pyduf** [comment]
> Validation ! Thanks for the split, better to spot differences now :) added some new comments, mainly guidelines / organization

Thanks for the review. I have made the changes.

File: addons/event/models/event_registration.py [hunk 8]
```diff
@@ -47,6 +47,9 @@ class EventRegistration(models.Model):
         ('draft', 'Unconfirmed'), ('cancel', 'Cancelled'),
         ('open', 'Confirmed'), ('done', 'Attended')],
         string='Status', default='draft', readonly=True, copy=False, tracking=True)
+    # event related field added for search
```
Comment (awa-odoo) [inline addons/event/models/event_registration.py:50]: I would move those 2 under "event_end_date", sine those fields are already related to the event_id, and are under the "organisation" category (which fits well with organizer and responsible).

File: addons/event/views/event_event_views.xml [hunk 9]
```diff
@@ -11,6 +11,14 @@
                 </header>
                 <sheet>
                     <div class="oe_button_box" name="button_box" groups="base.group_user">
+                        <button name="%(event.event_registration_action_stats_from_event)d"
+                                type="action"
+                                class="oe_stat_button"
+                                icon="fa-line-chart">
```
Comment (awa-odoo) [inline addons/event/views/event_event_views.xml:17]: Let's put all those on the same line.

File: addons/event/views/event_registration_views.xml [hunk 10]
```diff
@@ -246,18 +249,41 @@
...
+                    <filter string="Medium" name="group_by_utm_medium_id" domain="[]"
+                            context="{'group_by': 'utm_medium_id'}"/>
+                    <filter string="Source" name="group_by_utm_source_id" domain="[]"
+                            context="{'group_by': 'utm_source_id'}"/>
                </group>
             </search>
         </field>
     </record>
 
+    <!-- Search view specific for an event so remove event related fields and filters -->
```
Comment (awa-odoo) [inline addons/event/views/event_registration_views.xml:263]: Maybe rephrase a bit?

Sugg:

"Search view typically used when coming from a specific event, meaning we don't need event-related fields"

File: addons/event/views/event_registration_views.xml [hunk 11]
```diff
@@ -327,6 +353,29 @@
         </field>
     </record>
 
+    <record id="event_registration_action_stats_from_event" model="ir.actions.act_window">
+        <field name="name">Registration statistics</field>
+        <field name="res_model">event.registration</field>
+        <field name="view_mode">graph,pivot,kanban,tree,form</field>
+        <field name="domain">[('event_id', '=', active_id)]</field>
+        <field name="context">
+            {
```
Comment (awa-odoo) [inline addons/event/views/event_registration_views.xml:362]: Can we use the same syntax as the other one? Meaning the opening bracket is on the same line as the field tag?

File: addons/event/views/event_registration_views.xml [hunk 12]
```diff
@@ -327,6 +353,29 @@
...
+            'default_event_id': active_id,
+            'search_default_group_by_create_date_day': 1,
+            'search_default_status': 2,
+            'registration_view_hide_group_by_create_date_week': 1,
```
Comment (awa-odoo) [inline addons/event/views/event_registration_views.xml:366]: And these should have an extra indentation level if we refer to the other one you modified.

**Review by awa-odoo — COMMENTED** [review]
Hello @pyduf 

Nice work overall, I added some light polishing comments.

Can I also ask you to modify your commit message a bit?
It goes into a lot of details currently.
If you want to go into details that's fine but you should at least have a general purpose at the beginning.

e.g:

> PURPOSE
> 
> Globally ease event registrations followup by improving the related search
> views and by allowing a quick access to statistics.
> 
> SPECS
> 
> [and then your detailed changes]

That gives the reader a general idea of why we made those changes, as he can see "what" by reading the diff.

Not a big deal, but it helps having a nice commit history.

Thanks & cheers!

File: addons/event/models/event_registration.py (hunk 8 above)
Comment (pyduf) [reply addons/event/models/event_registration.py]: Thanks, indeed, I have also removed my comment; other fields are not commented and have probably the same purpose

File: addons/event/views/event_registration_views.xml (hunk 10 above)
Comment (pyduf) [reply addons/event/views/event_registration_views.xml]: Yes better, thanks

File: addons/event/models/event_registration.py [hunk 13]
```diff
@@ -40,6 +40,8 @@ class EventRegistration(models.Model):
         readonly=False, store=True)
     event_begin_date = fields.Datetime(string="Event Start Date", related='event_id.date_begin', readonly=True)
     event_end_date = fields.Datetime(string="Event End Date", related='event_id.date_end', readonly=True)
+    event_organizer_id = fields.Many2one(string='Event organizer', related='event_id.organizer_id', readonly=True)
+    event_user_id = fields.Many2one(string='Event responsible', related='event_id.user_id', readonly=True)
```
Comment (awa-odoo) [inline addons/event/models/event_registration.py:44]: We usually capitalize those
("Event Organizer" and "Event Responsible").

**Review by awa-odoo — COMMENTED** [review]
Hello @pyduf 

One last tiny remark and then we are good to go.

Thanks & cheers!

File: addons/event/models/event_registration.py (hunk 13 above)
Comment (pyduf) [reply addons/event/models/event_registration.py]: Done, thanks

**Comment by awa-odoo** [comment]
@robodoo retry
